rust_src/src/symbols.rs

```diff
@@ -487,6 +489,48 @@ pub fn local_variable_p(mut symbol: LispSymbolRef, buffer: LispBufferOrCurrent)
     }
 }
 
+/// Return a value indicating where VARIABLE's current binding comes from.
+/// If the current binding is buffer-local, the value is the current buffer.
+/// If the current binding is global (the default), the value is nil.
+#[lisp_fn]
+pub fn variable_binding_locus(mut symbol: LispSymbolRef) -> LispObject {
+    // Make sure the current binding is actually swapped in.
+    unsafe {
+        symbol.find_value();
+    }
+    symbol = symbol.get_indirect_variable();
+
+    fn localized_handler(sym: LispSymbolRef) -> LispObject {
+        // For a local variable, record both the symbol and which
+        // buffer's or frame's value we are saving.
+        let blv = unsafe { sym.get_blv() };
+
+        if local_variable_p(sym, LispBufferOrCurrent::Current) {
+            current_buffer().into()
```

Comment:
```suggestion
            current_buffer()
```
Clippy identifies this as an unneeded conversion.